otlp/trace_test.go

```diff
@@ -779,7 +779,7 @@ func TestGetSampleRateConversions(t *testing.T) {
 	}
 }
 
-func TestMissingServiceNameAttributeReturnsError(t *testing.T) {
+func TestMissingServiceNameAttributeUsesDefault(t *testing.T) {
```

Comment:
All of these tests could be combined into a single test with different test cases.

For example
```
testCases := []interface{
	name		string
	resource		Resource
	serviceName	string
}{
	{
		name: "nil resource",
		resource: nil,
		serviceName: "unknown_service"
	},
	{
		name: "empty string",
		resource: &resource.Resource{
			Attributes: []*common.KeyValue{{
				Key: "service.name",
				Value: &common.AnyValue{
					Value: &common.AnyValue_StringValue{StringValue: ""},
				},
			}},
		},
		serviceName: "unknown_service"
	},
	// .. bool, int, etc
}

for _, tc := range testCases {
	t.run(tc.name, func (t *testing.T) {
		// setup the test using tc.resource
		req := &collectortrace.ExportTraceServiceRequest{
			ResourceSpans: []*trace.ResourceSpans{{
				Resource: tc.resource,
				InstrumentationLibrarySpans: []*trace.InstrumentationLibrarySpans{{
					Spans: []*trace.Span{{
						TraceId: test.RandomBytes(16),
						SpanId:  test.RandomBytes(8),
						Name:    "test_span_a",
					}},
				}},
			}},
		}
		// test with tc.serviceName
		assert.Equal(t, "unknown_service", batch.Dataset)
	})
}
``